src/span.cpp

```diff
@@ -148,8 +148,14 @@ void Span::FinishWithOptions(
   }
   tag = span_->meta.find(datadog_error_tag);
   if (tag != span_->meta.end()) {
-    span_->error = std::stoi(tag->second);
-    span_->meta.erase(tag);
+    // tag->second is the JSON-serialized value of the variadic type given to SetTag. If it's
+    // clearly falsey then set the error flag.
+    if (tag->second == "0" || tag->second == "" || tag->second == "false") {
```

Comment:
This makes me a little bit sad.

Reply:
Me too! I wasn't sure of the best way in C++ to take variant_type and turn it into a bool
I figured there were more possible "things that are probably an error" than otherwise, and that nearly always the usage of setting the error tag would be to set an error

So I wanted to err on the side of making it an error.

I also though about exactly following JS "truthy" logic, since the string result comes from JSON-ifying the variant_type (so also "undefined", "null" strings falsey), but figured that was getting a bit much.